### ISSUE TITLE
EmmyLua(unused) How to specify rule suppression pattern

### ISSUE THREAD

**Issue by vaderkos** [EmmyLuaLs/emmylua-analyzer-rust#737]
Is there a way to specify the regex that suppresses unused fields.
We are using a pattern where private/protected variables are prefixed with underscore with collides with the `EmmyLua(unused)` rule making unused variables not triggering the rule.

```lua

local _lua_setmetatable = setmetatable
local _lua_table_remove = table.remove
```

Nevertheless these variables are unused they're are not triggering unused rule.

Can I, for example in rule configuration change the pattern to suppress `EmmyLua(unused)`?

**Comment by CppCXY** [comment]
This rule comes from Rust, and I think it fits the general usage of Lua, so I will not add a configuration here. I suggest you follow this rule.

**Comment by vaderkos** [comment]
@CppCXY this would make it not flexible enough. Hundreds of files to rewrite in the codebase, which would rather lead to abandoning using this analyzer.

**Comment by vaderkos** [comment]
Because https://modiki.civfanatics.com/index.php/Specificities_of_the_Lua_implementation_in_Civ5

**Comment by CppCXY** [comment]
I will not support incorrect coding practices. I suggest you either modify your code to follow standard conventions, or simply ignore the EmmyLua unused check. You can achieve this with other tools, such as the luacheck plugin.

If you learn Rust, I suggest directly modifying the source code. All of these can be easily changed to fit your needs; I just don't want to do it myself.
